app/services/data_exports/data_storage.rb

```diff
@@ -0,0 +1,125 @@
+require 'fileutils'
+require 'securerandom'
+
+module DataExports
+  class NoStoredFilesFoundError < StandardError; end
+
+  class DataStorage
+    # Public: downloads all transcription files for a given transcription
+    # returns path to zip file
+    def zip_transcription_files(transcription)
+      Dir.mktmpdir { |directory_path|
+        if transcription.files.attached?
+          transcription_folder = download_transcription_files(transcription, directory_path)
+          yield zip_files(directory_path, transcription_folder)
+        else
+          raise NoStoredFilesFoundError.new("No stored files found for transcription with id '#{transcription.id}'")
+        end
+      }
+    end
+
+    # Public : downloads all transcription group files for a given group
+    # returns path to zip file
+    def zip_group_files(transcriptions)
+      Dir.mktmpdir { |directory_path|
+        group_folder = File.join(directory_path, "group_#{transcriptions.first.group_id}")
+        FileUtils.mkdir_p(group_folder)
+
+        metadata_file_gen = AggregateMetadataFileGenerator.new(group_folder)
+        metadata_file_gen.generate_group_file(transcriptions)
+
+        transcriptions.each do |t|
+          download_transcription_files(t, group_folder) if t.files.attached?
+        end
+
+        yield zip_files(directory_path, group_folder)
+      }
+    end
+
+    # Public : downloads all files for a given workflow
+    # returns path to zip file
+    def zip_workflow_files(workflow)
```

Comment:
Not blocking: It seems that the public methods in this class aren't really related but really are different strategies for handling file storage use cases (reading the zip prefix on each method implies common behaviour, i.e. zip, and the different cases are the method name suffixes post zip)

I think you could split out to each method to a different `DataStorage` class (possibly with a common base class for shared behaviour). 

I'd be happy to chat about this (and about Single Responsiblity Principle), again not blocking but this class seems a dumping ground for data storage behaviour that may continue to grow.

Reply:
Yeah it'd be nice to talk about this! I'll dm you.

Reply:
Making an issue for this as a nice-to-have refactor.